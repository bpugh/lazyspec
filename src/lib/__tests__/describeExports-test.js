const fs = require('fs');

const describeExports = require('../describeExports');

function expectDescription(moduleName) {
  const unitPath = `spec/fixtures/${moduleName}.js`;
  const src = fs.readFileSync(unitPath).toString();
  const description = describeExports(unitPath, src);

  expect(description).toMatchSnapshot();
}

describe('describeExports', () => {
  it('should describe simple exports', () => {
    expectDescription('esExports');
  });

  it('should describe an ES5 component', () => {
    expectDescription('CreateClass');
  });

  it('should describe an ES6 component', () => {
    expectDescription('DefaultableComponent');
  });

  it('should describe an ES6 with Flow component', () => {
    expectDescription('FlowComponent');
  });

  it('should describe a stateless function component', () => {
    expectDescription('StatelessFunction');
  });

  it('should describe multi-exports', () => {
    expectDescription('MultipleComponents');
  });
});
