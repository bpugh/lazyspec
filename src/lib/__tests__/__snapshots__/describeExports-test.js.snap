// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describeExports should describe multi-exports 1`] = `
Object {
  "exportsDefault": true,
  "hasExports": true,
  "moduleName": "spec/fixtures/MultipleComponents.js",
  "namedExportNames": Array [
    "Bold",
  ],
}
`;

exports[`describeExports should describe simple exports 1`] = `
Object {
  "exportsDefault": true,
  "hasExports": true,
  "moduleName": "spec/fixtures/esExports.js",
  "namedExportNames": Array [
    "refConst",
    "refFunc",
    "inlineConst",
    "inlineFunc",
  ],
}
`;
